FinancialTransactionsApi/V1/Controllers/FinancialTransactionsApiController.cs

```diff
@@ -68,16 +66,12 @@ public class FinancialTransactionsApiController : BaseController
         [ProducesResponseType(typeof(BaseErrorResponse), StatusCodes.Status500InternalServerError)]
         [HttpGet]
         [Route("{id}")]
-        public async Task<IActionResult> Get([FromHeader(Name = "x-correlation-id")] string correlationId, [FromRoute] Guid id, [FromQuery] Guid targetId)
+        public async Task<IActionResult> Get([FromRoute] Guid id)
         {
-            var transaction = await _getByIdUseCase.ExecuteAsync(id, targetId).ConfigureAwait(false);
 
-            if (transaction == null)
-            {
-                return NotFound(new BaseErrorResponse((int) HttpStatusCode.NotFound, "No transaction by provided Id cannot be found!"));
-            }
+            ResponseWrapper<TransactionResponse> transaction = await _getByIdUseCase.ExecuteAsync(id).ConfigureAwait(false);
 
-            return Ok(transaction);
+            return (transaction.IsEmpty) ? NotFound(id) : Ok(transaction.Value);
```

Comment:
Please do not use a ternary. Use normal if-else statement to make this more readable and to make clearer the intention of the Controller logic

```
if (transaction.IsEmpty)
{
    return NotFound("No transaction exists for this id: {id}, id);
}

return ok(tranaction.Value);
```